From 8a353689559cd864647d39a05ef449e423a07fdd Mon Sep 17 00:00:00 2001
From: Elvis Dowson <elvis.dowson@gmail.com>
Date: Wed, 22 Aug 2012 01:38:03 +0400
Subject: [PATCH 1/3] Xilinx: Update IP Core support for Xilinx ISE 14.1.

This commit updates linux kernel support for the following IP Cores
shipped with Xilinx ISE 14.1:
xlnx,xps-sysace-1.01.a
xlnx,xps-iic-2.02.a
xlnx,xps-ethernetlite-4.00.a
xlnx,xps-ll-temac-2.03.a
xlnx,xps-uartlite-1.02.a

This commit also updates PowerPC support for the following IP Cores
shipped with Xilinx ISE 14.1:
xps-uartlite-1.02.a
xlnx,plb-v46-1.05.a
xlnx,xps-intc-2.01.a

Signed-off-by: Elvis Dowson <elvis.dowson@gmail.com>
---
 arch/powerpc/boot/serial.c                         |    4 +++-
 arch/powerpc/platforms/44x/virtex.c                |    3 +++
 arch/powerpc/sysdev/xilinx_intc.c                  |    2 ++
 drivers/block/xsysace.c                            |    1 +
 drivers/i2c/algos/xilinx_iic/i2c-algo-xilinx.c     |    2 ++
 drivers/net/ethernet/xilinx/xilinx_emaclite.c      |    2 ++
 .../ethernet/xilinx/xilinx_lltemac/xlltemac_main.c |    4 ++++
 drivers/tty/serial/uartlite.c                      |    2 ++
 8 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/arch/powerpc/boot/serial.c b/arch/powerpc/boot/serial.c
index f2156f0..6f901dc 100644
--- a/arch/powerpc/boot/serial.c
+++ b/arch/powerpc/boot/serial.c
@@ -130,7 +130,9 @@ int serial_console_init(void)
 	else if (dt_is_compatible(devp, "fsl,mpc5200-psc-uart"))
 		rc = mpc5200_psc_console_init(devp, &serial_cd);
 	else if (dt_is_compatible(devp, "xlnx,opb-uartlite-1.00.b") ||
-		 dt_is_compatible(devp, "xlnx,xps-uartlite-1.00.a"))
+		dt_is_compatible(devp, "xlnx,xps-uartlite-1.00.a") ||
+		dt_is_compatible(devp, "xlnx,xps-uartlite-1.01.a") ||
+		dt_is_compatible(devp, "xlnx,xps-uartlite-1.02.a"))
 		rc = uartlite_console_init(devp, &serial_cd);
 
 	/* Add other serial console driver calls here */
diff --git a/arch/powerpc/platforms/44x/virtex.c b/arch/powerpc/platforms/44x/virtex.c
index cf96cca..eb208d3 100644
--- a/arch/powerpc/platforms/44x/virtex.c
+++ b/arch/powerpc/platforms/44x/virtex.c
@@ -25,6 +25,9 @@ static struct of_device_id xilinx_of_bus_ids[] __initdata = {
 	{ .compatible = "simple-bus", },
 	{ .compatible = "xlnx,plb-v46-1.00.a", },
 	{ .compatible = "xlnx,plb-v46-1.02.a", },
+	{ .compatible = "xlnx,plb-v46-1.03.a", },
+	{ .compatible = "xlnx,plb-v46-1.04.a", },
+	{ .compatible = "xlnx,plb-v46-1.05.a", },
 	{ .compatible = "xlnx,plb-v34-1.01.a", },
 	{ .compatible = "xlnx,plb-v34-1.02.a", },
 	{ .compatible = "xlnx,opb-v20-1.10.c", },
diff --git a/arch/powerpc/sysdev/xilinx_intc.c b/arch/powerpc/sysdev/xilinx_intc.c
index 6183799..a115a20 100644
--- a/arch/powerpc/sysdev/xilinx_intc.c
+++ b/arch/powerpc/sysdev/xilinx_intc.c
@@ -267,6 +267,8 @@ static inline void xilinx_i8259_setup_cascade(void) { return; }
 static struct of_device_id xilinx_intc_match[] __initconst = {
 	{ .compatible = "xlnx,opb-intc-1.00.c", },
 	{ .compatible = "xlnx,xps-intc-1.00.a", },
+	{ .compatible = "xlnx,xps-intc-2.00.a", },
+	{ .compatible = "xlnx,xps-intc-2.01.a", },
 	{}
 };
 
diff --git a/drivers/block/xsysace.c b/drivers/block/xsysace.c
index 6e9a86b..8b2ee12 100644
--- a/drivers/block/xsysace.c
+++ b/drivers/block/xsysace.c
@@ -1194,6 +1194,7 @@ static const struct of_device_id ace_of_match[] __devinitconst = {
 	{ .compatible = "xlnx,opb-sysace-1.00.b", },
 	{ .compatible = "xlnx,opb-sysace-1.00.c", },
 	{ .compatible = "xlnx,xps-sysace-1.00.a", },
+	{ .compatible = "xlnx,xps-sysace-1.01.a", },
 	{ .compatible = "xlnx,sysace", },
 	{},
 };
diff --git a/drivers/i2c/algos/xilinx_iic/i2c-algo-xilinx.c b/drivers/i2c/algos/xilinx_iic/i2c-algo-xilinx.c
index 4f3c599..c59b6d0 100644
--- a/drivers/i2c/algos/xilinx_iic/i2c-algo-xilinx.c
+++ b/drivers/i2c/algos/xilinx_iic/i2c-algo-xilinx.c
@@ -554,6 +554,8 @@ static int __devinit xilinx_iic_setup(
 /* Match table for of_platform binding */
 static struct of_device_id __devinitdata xilinx_iic_of_match[] = {
 	{ .compatible = "xlnx,xps-iic-2.00.a", },
+	{ .compatible = "xlnx,xps-iic-2.01.a", },
+	{ .compatible = "xlnx,xps-iic-2.02.a", },
 	{},
 };
 MODULE_DEVICE_TABLE(of, xilinx_iic_of_match);
diff --git a/drivers/net/ethernet/xilinx/xilinx_emaclite.c b/drivers/net/ethernet/xilinx/xilinx_emaclite.c
index 79013e5..91fb8e0 100644
--- a/drivers/net/ethernet/xilinx/xilinx_emaclite.c
+++ b/drivers/net/ethernet/xilinx/xilinx_emaclite.c
@@ -1289,6 +1289,8 @@ static struct of_device_id xemaclite_of_match[] __devinitdata = {
 	{ .compatible = "xlnx,xps-ethernetlite-2.00.a", },
 	{ .compatible = "xlnx,xps-ethernetlite-2.01.a", },
 	{ .compatible = "xlnx,xps-ethernetlite-3.00.a", },
+	{ .compatible = "xlnx,xps-ethernetlite-3.01.a", },
+	{ .compatible = "xlnx,xps-ethernetlite-4.00.a", },
 	{ /* end of list */ },
 };
 MODULE_DEVICE_TABLE(of, xemaclite_of_match);
diff --git a/drivers/net/ethernet/xilinx/xilinx_lltemac/xlltemac_main.c b/drivers/net/ethernet/xilinx/xilinx_lltemac/xlltemac_main.c
index e008795..b05e996 100644
--- a/drivers/net/ethernet/xilinx/xilinx_lltemac/xlltemac_main.c
+++ b/drivers/net/ethernet/xilinx/xilinx_lltemac/xlltemac_main.c
@@ -3595,6 +3595,7 @@ static struct of_device_id xtenet_fifo_of_match[] = {
 	{ .compatible = "xlnx,xps-ll-fifo-1.00.a", },
 	{ .compatible = "xlnx,xps-ll-fifo-1.00.b", },
 	{ .compatible = "xlnx,xps-ll-fifo-1.01.a", },
+	{ .compatible = "xlnx,xps-ll-fifo-1.02.a", },
 	{ /* end of list */ },
 };
 
@@ -3746,6 +3747,9 @@ static struct of_device_id xtenet_of_match[] = {
 	{ .compatible = "xlnx,xps-ll-temac-1.00.b", },
 	{ .compatible = "xlnx,xps-ll-temac-1.01.a", },
 	{ .compatible = "xlnx,xps-ll-temac-1.01.b", },
+	{ .compatible = "xlnx,xps-ll-temac-2.00.a", },
+	{ .compatible = "xlnx,xps-ll-temac-2.02.a", },
+	{ .compatible = "xlnx,xps-ll-temac-2.03.a", },
 	{ /* end of list */ },
 };
 
diff --git a/drivers/tty/serial/uartlite.c b/drivers/tty/serial/uartlite.c
index 6cd4143..dd0664d 100644
--- a/drivers/tty/serial/uartlite.c
+++ b/drivers/tty/serial/uartlite.c
@@ -566,6 +566,8 @@ static int __devexit ulite_release(struct device *dev)
 static struct of_device_id ulite_of_match[] __devinitdata = {
 	{ .compatible = "xlnx,opb-uartlite-1.00.b", },
 	{ .compatible = "xlnx,xps-uartlite-1.00.a", },
+	{ .compatible = "xlnx,xps-uartlite-1.01.a", },
+	{ .compatible = "xlnx,xps-uartlite-1.02.a", },
 	{}
 };
 MODULE_DEVICE_TABLE(of, ulite_of_match);
-- 
1.7.9.5

